3Dmol/glviewer.js

```diff
@@ -1055,6 +1055,31 @@ $3Dmol.GLViewer = (function() {
         this.selectedAtoms = function(sel) {
             return getAtomsFromSel(sel);
         };
+
+        /**
+        * Returns valid values for the specified attribute in the given selection
+        * @function $3Dmol.GlViewer#getUniqueValues 
+        * @param {string} attribute
+        * @param {AtomSelectionSpec} sel
+        * @return {Array.<Object>}
+        *
+        */
+        this.getUniqueValues = function(attribute, sel){
+            if (typeof (sel) === "undefined")
+                sel = {};            
+            var atoms = getAtomsFromSel(sel);
+            var values = [];
```

Comment:
Don't use lists for membership testing.  Each test is linear.  Use a dictionary.